src/models/core/SearchConfig.ts

```diff
@@ -1,11 +1,33 @@
 import { Endpoints } from './Endpoints';
 import { Visitor } from './Visitor';
+import { Environment } from './Environment';
+import { CloudRegion } from './CloudRegion';
+
+/**
+ * The configuration options for {@link provideEndpoints}.
+ *
+ * @public
+ */
+export interface ServingConfig {
+  /**
+   * {@inheritDoc Environment}
+   *
+   * @public
+   */
+  environment?: Environment,
+  /**
+   * {@inheritDoc CloudRegion}
+   *
+   * @public
+   */
+  cloudRegion?: CloudRegion
+}
 
 /**
  * The base configuration options for {@link SearchCore}.
  * @public
  */
-export interface BaseSearchConfig {
+export interface BaseSearchConfig extends ServingConfig {
```

Comment:
even though BaseSearchConfig extends ServingConfig, it may be helpful to include the environment and cloudRegion properties in the BaseSearchConfig interface so that they appear in the generated docs.

Reply:
I think maybe if we `@link ServingConfig` in the json doc or do something similar it will pull the docs in if ServingConfig is exported

Reply:
The generated docs link to the ServingConfig it extends it, but it doesn't include its properties on the BaseConfig interface doc page. In the past on Slapshot we made the decision to duplicate the properties for the sake of clearer docs

Reply:
Updated